app/src/main/java/com/github/sdpteam15/polyevents/MainActivity.kt

```diff
@@ -2,40 +2,51 @@ package com.github.sdpteam15.polyevents
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
-import android.view.Menu
 import android.widget.ArrayAdapter
+import com.google.android.material.bottomnavigation.BottomNavigationView
+
+import android.view.Menu
 import android.widget.SearchView
 import androidx.fragment.app.Fragment
 import com.github.sdpteam15.polyevents.fragments.*
-import com.google.android.material.bottomnavigation.BottomNavigationView
+import com.github.sdpteam15.polyevents.helper.HelperFunctions
 
 class MainActivity : AppCompatActivity() {
+    companion object{
+        private val mapFragment:MutableMap<Int, Fragment> = HashMap()
+        //make the fragments available from outside of the activity and instantiate only once
+        val fragments:Map<Int, Fragment>
+            get(){
+                if(mapFragment.isEmpty()){
+                    mapFragment[R.id.ic_home] = HomeFragment()
+                    mapFragment[R.id.ic_map] = MapsFragment()
+                    mapFragment[R.id.ic_list] = ListFragment()
+                    mapFragment[R.id.ic_login] = LoginFragment()
+                    mapFragment[R.id.ic_more] = MoreFragment()
+                    mapFragment[R.id.id_fragment_profile] = ProfileFragment()
+                }
+                //return type immutable
+                return HashMap<Int, Fragment>(mapFragment)
+            }
```

Comment:
mapFragment could init as null and then be tested as null instead of testing isEmpty in the get method.
We then initialize it in the if block.

Reply:
Also modified